Gems/GradientSignal/Code/Source/Editor/EditorGradientBakerComponent.cpp

```diff
@@ -16,6 +21,176 @@
+            AZ_Error("GradientBaker", false, "Failed to write out gradient baked image to path: %s",
+                absolutePath.c_str());
+            return;
+        }
+
+        OIIO::ImageSpec spec(imageResolutionX, imageResolutionY, channels, pixelFormat);
+        outputImage->open(absolutePath.c_str(), spec);
+
+        const AZ::Vector3 inputBoundsCenter = m_inputBounds.GetCenter();
+        const AZ::Vector3 inputBoundsExtentsOld = m_inputBounds.GetExtents();
+        m_inputBounds =
+            AZ::Aabb::CreateCenterRadius(inputBoundsCenter, AZ::GetMax(inputBoundsExtentsOld.GetX(), inputBoundsExtentsOld.GetY()) / 2.0f);
+
+        const AZ::Vector3 inputBoundsStart =
+            AZ::Vector3(m_inputBounds.GetMin().GetX(), m_inputBounds.GetMin().GetY(), inputBoundsCenter.GetZ());
+        const AZ::Vector3 inputBoundsExtents = m_inputBounds.GetExtents();
+        const float inputBoundsExtentsX = inputBoundsExtents.GetX();
+        const float inputBoundsExtentsY = inputBoundsExtents.GetY();
+
+        // When sampling the gradient, we can choose to either do it at the corners of each texel area we're sampling, or at the center.
+        // They're both correct choices in different ways.  We're currently choosing to do the corners, which makes scaledTexelOffset = 0,
+        // but the math is here to make it easy to change later if we ever decide sampling from the center provides a more intuitive
+        // image.
+        constexpr float texelOffset = 0.0f; // Use 0.5f to sample from the center of the texel.
+        const AZ::Vector3 scaledTexelOffset(
+            texelOffset * inputBoundsExtentsX / static_cast<float>(imageResolutionX),
+            texelOffset * inputBoundsExtentsY / static_cast<float>(imageResolutionY), 0.0f);
+
+        // Scale from our image size space (ex: 256 pixels) to our bounds space (ex: 16 meters)
+        const AZ::Vector3 pixelToBoundsScale(
+            inputBoundsExtentsX / static_cast<float>(imageResolutionX), inputBoundsExtentsY / static_cast<float>(imageResolutionY), 0.0f);
+
+        for (int y = 0; y < imageResolutionY; ++y)
+        {
+            for (int x = 0; x < imageResolutionX; ++x)
+            {
+                // Invert world y to match axis.  (We use "imageBoundsY- 1" to invert because our loop doesn't go all the way to
+                // imageBoundsY)
+                AZ::Vector3 uvw(static_cast<float>(x), static_cast<float>((imageResolutionY - 1) - y), 0.0f);
+
+                GradientSampleParams sampleParams;
+                sampleParams.m_position = inputBoundsStart + (uvw * pixelToBoundsScale) + scaledTexelOffset;
+
+                bool inBounds = true;
+                LmbrCentral::ShapeComponentRequestsBus::EventResult(
+                    inBounds, m_boundsEntityId, &LmbrCentral::ShapeComponentRequestsBus::Events::IsPointInside, sampleParams.m_position);
+
+                float sample = inBounds ? m_configuration.m_gradientSampler.GetValue(sampleParams) : 0.0f;
```

Comment:
FWIW, this could potentially be made a lot faster by breaking this into a few steps:
1. Generate a full set of input positions (after filtering the list with IsPointInside), and an equivalent list of indices.
2. Call gradientSampler.GetValues() once with the full set of input positions.
3. Process all the output values into writeImage.

There's a much heavier memory cost to it, along with additional allocation / deallocation time, but GetValues() vs GetValue() will be significantly faster, especially in long gradient chains.

It might be worth trying out and measuring to see if it works out better, and if so, submitting as a separate PR.

Reply:
Good call, I've added a line to the AC of my follow-up task to investigate this. I would just roll it into this PR but I think the diff might not show well since all the baking logic is being moved to the job.